File: LibraryAggregator.DataLayer/Entities/OperatingModes.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace LibraryAggregator.DataLayer.Entities
{
    public class OperatingModes
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int OperatingModeId { get; set; }
        public string? Monday { get; set; }
        public string? Tuesday { get; set; }
        public string? Wednesday { get; set; }
        public string? Thursday { get; set; }
        public string? Friday { get; set; }
        public string? Saturday { get; set; }
        public string? Sunday { get; set; }
        public int? LibraryId { get; set; }
        public Library Libraries { get; set; }
    }
}

File: LibraryAggregator.WebClient/src/app/models/operatingModes.ts
import { Library } from "./library";

export interface OperatingModes {
    operatingModesId: number;
    monday?: string;
    tuesday?: string;
    wednesday?: string;
    thursday?: string;
    friday?: string;
    saturday?: string;
    sunday?: string;
    libraryId: number;
    library: Library;
} 